File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        zeroCount = 0
        for num in nums:
            if num == 0: 
                zeroCount+=1
                continue
            prod *= num
        ans = []
        for num in nums:
            if zeroCount > 1:
                ans.append(0)
                continue;
            elif zeroCount == 1:
                if num == 0:
                    ans.append(prod)
                else: ans.append(0)
            else:    
                ans.append(int(prod/num))
        return ans

File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n = nums.size();
        set<vector<int>> temp;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                for(int k=j+1;k<n;k++){
                    if(nums[i]+nums[j]+nums[k] == 0){
                        vector<int> t;
                        t = {nums[i], nums[j], nums[k]};
                        sort(t.begin(), t.end());
                        temp.insert(t);
                    }
                }
            }
        }
        vector<vector<int>> ans;
        for(auto it: temp) ans.push_back(it);
        return ans;
    }
};
